File: Models/Transaction.swift
import Foundation

struct Transaction: Identifiable, Codable {

    // MARK: - Core Identity
    let id: UUID

    // MARK: - Transaction Details
    var title: String              // Merchant or description
    var notes: String
    var amount: Double
    var date: Date

    // MARK: - Classification
    var category: Category
    var isIncome: Bool

    // MARK: - Recurring
    var recurringRule: RecurringRule?

    // MARK: - Initialiser
    init(
        id: UUID = UUID(),
        title: String,
        notes: String = "",
        amount: Double,
        date: Date,
        category: Category,
        isIncome: Bool,
        recurringRule: RecurringRule? = nil
    ) {
        self.id = id
        self.title = title
        self.notes = notes
        self.amount = amount
        self.date = date
        self.category = category
        self.isIncome = isIncome
        self.recurringRule = recurringRule
    }
}

File: ledgerly-android/app/src/main/java/com/ledgerly/data/models/Transaction.kt
package com.ledgerly.data.models

import java.time.LocalDate

data class Transaction(
    val id: String,
    val title: String,
    val notes: String = "",
    val amount: Double,
    val date: LocalDate,
    val category: Category,
    val isIncome: Boolean,
    val recurringRule: RecurringRule? = null
)